source/adapters/cuda/command_buffer.cpp

```diff
@@ -71,16 +71,42 @@ ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
   }
 }
 
-ur_exp_command_buffer_command_handle_t_::
-    ur_exp_command_buffer_command_handle_t_(
-        ur_exp_command_buffer_handle_t CommandBuffer, ur_kernel_handle_t Kernel,
-        CUgraphNode Node, CUDA_KERNEL_NODE_PARAMS Params, uint32_t WorkDim,
-        const size_t *GlobalWorkOffsetPtr, const size_t *GlobalWorkSizePtr,
-        const size_t *LocalWorkSizePtr)
-    : CommandBuffer(CommandBuffer), Kernel(Kernel), Node(Node), Params(Params),
-      WorkDim(WorkDim), RefCountInternal(1), RefCountExternal(1) {
-  CommandBuffer->incrementInternalReferenceCount();
+std::unique_ptr<ur_event_handle_t_>
+ur_exp_command_buffer_handle_t_::addSignalNode(CUgraphNode DepNode,
+                                               CUgraphNode &SignalNode) {
+  CUevent Event;
+  UR_CHECK_ERROR(cuEventCreate(&Event, CU_EVENT_DEFAULT));
```

Comment:
In the normal non graph API, we only record events at the beginning of an event if profiling is enabled. Events recorded afterwards for dependency analysis are always recorded if profiling is not being used. Should we have some sort of check here that checks if profiling is enabled? We can skip recording an extra event before each node this way

Reply:
SYCL-Graphs doesn't support profiling on the node level (i.e. inside a graph rather than whole graph submission) yet, so even if profiling was enabled we don't have a way to express back up the stack any of the counters results from per node events.

One of the intents of this PR is to provide a SYCL functional equivalent of `cudaGraphAddEventRecordNode`, we decided not to create a separate UR entry-point (e.g. `urCommandBufferAppendEventSignal`) for this, nor a separate SYCL-Graph node type (e.g `graph.add_signal_node()`) and instead add an extra parameter to existing UR/SYCL-Graph command entry-points, which when non-null would result in the creation of the extra `cudaGraphAddEventRecordNode` node.

Reply:
Fair enough. That sounds reasonable to me.